### DISCUSSION TITLE
Is there possibility to assign people from Gitlab project members to failing MRs?

### DISCUSSION THREAD

**Discussion by marcinpz** [renovatebot/renovate#11437] [Request Help]
I have default Renovate config for our group / namespace in Gitlab. I would like Renovate assigned people from project members to Merge Requests if they fail to pass the tests.
I tried option `assigneesFromCodeOwners` but as I understand it uses special files `CODEOWNERS` which we don't use.
I don't want to specify `assignees` for each project directly.

**Comment by viceice** [comment]
this is not possible

**Reply by marcinpz** [reply]
Wasn't it nice feature? How do think?

**Reply by viceice** [reply]
I think this would be hard to implement, as our current api's don't support such things.

Please clarify your feature request with some more details how it shoud work.

**Reply by marcinpz** [reply]
So basically I thought that `assigneesFromCodeOwners` option means that project owners (taken from members list) will be used (I knew nothing about `CODEOWNERS` file). 
So if there is no `CODEOWNERS` file for the project then [Gitlab project members API](https://docs.gitlab.com/ee/api/members.html#list-all-members-of-a-group-or-project) could be used instead to notify 'owners' about failing MR. This option could be enabled by default so owners knew there is an issue.
I work with Gitlab for 2 years now and did not know there could be `CODEOWNERS` file. It is Gitlab feature I've never used. I wonder why project members are not used for this purpose...

**Reply by rarkins** [reply]
The challenge is how to handle the scenario where we would assign someone on failure but then assignees get changed. We don't really know when the "first" time we run after a failure is. e.g. let's say it's like this:
- Create MR. No assignees
- MR pipeline failure
- Renovate runs, adds assignees due to failure
- Assignees unassign themselves
- Renovate runs, adds assignees back?

We could possibly have a rule which says "assign on failure only if no one else is assigned" and that might cover it because it would only have a problem if everybody was unassigned and nobody else was assigned.

**Reply by marcinpz** [reply]
And how it works currently with `CODEOWNERS` file? Assigned person can also unassign himself from MR.
Anyway if MR starts to be anoying because of these assigns people would probably try to fix it or close. And that is good.
